src/mono/browser/runtime/diagnostics/server_pthread/stream-queue.ts

```diff
@@ -68,22 +68,22 @@ export class StreamQueue {
     }
 
     private onWorkAvailable(this: StreamQueue /*,event: Event */): void {
-        const buf = Memory.getI32(this.buf_addr) as unknown as VoidPtr;
+        const buf = getI32(this.buf_addr) as unknown as VoidPtr;
```

Comment:
are these operations atomic? do we need to expose atomic read/write primitives from c as exports and use those here?

Reply:
i see storeI32 is defined further down and is atomic, but I'm concerned about get/set needing to at least be the equivalent of volatile.read/write

Reply:
I just refactored shape of existing code without changing what it does here.

Reply:
right, i only wondered since i saw atomics further down in the fn.